portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -8249,20 +8249,26 @@ private String _get18nErrorRedirect(
 			Set<Locale> availableLocales)
 		throws PortalException {
 
-		String virtualHostname = getVirtualHostname(
+		TreeMap<String, String> virtualHostnames = getVirtualHostnames(
 			themeDisplay.getLayoutSet());
 
-		if (Validator.isNull(virtualHostname)) {
+		String virtualHostname = "";
+
+		if (virtualHostnames.isEmpty()) {
 			Company company = themeDisplay.getCompany();
```

Comment:
Shouldn't the default case (list empty) be the value returned by `getVirtualHostName(themeDisplay.getLayoutSet())`?

Reply:
Also, can you extract this whole block into its own private method?
```
private String _getVirtualHostname(TreeMap<String, String>  virtualHostNames, ThemeDisplay themeDisplay) {
    if (virtualHostnames.isEmpty()) {
        return ...;
    }
    return virtualHostnames.firstKey();
}
```

Reply:
getVirtualHostname(themeDisplay.getLayoutset()) returns the first element of the get the virtualHostnames TreeMap. If the VurtalHostnames tree is empty the getVirtualhostname will return an empty string that's why 8255 had a verification for an empty virtualhostname.